### PR TITLE
feat: use terminal colors in the build system

### LINKED ISSUE #6974: Use terminal colors for all interesting paths/options/properties in the build system

### PR DESCRIPTION
**Pull request by sockmaster27**
Fixes #6974

### CODE REVIEW

**Comment by sockmaster27** [comment]
Release:

image

Running in folder mode:

image

Name of output file being occupied by something else (in this case some directories):

image

image

**Comment by sockmaster27** [comment]
Slightly related - It seems that the terminal is quickly cluttered by having all the `Building project...` type steps sticking around forever. Perhaps they should disappear again when the step is completed without errors?

**Comment by magnus-madsen** [comment]
I think `flix.toml` should also be highlighted.

**Comment by magnus-madsen** [comment]
> Slightly related - It seems that the terminal is quickly cluttered by having all the `Building project...` type steps sticking around forever. Perhaps they should disappear again when the step is completed without errors?

In the past we tried clearing things, but it _always_ ended up being confusing in edge cases.

File: main/src/ca/uwaterloo/flix/api/Bootstrap.scala [hunk 1]
```diff
@@ -501,7 +500,9 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
   /**
     * Builds a jar package for the project.
     */
-  def buildJar(): Validation[Unit, BootstrapError] = {
+  def buildJar(flix: Flix): Validation[Unit, BootstrapError] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Bootstrap.scala:503]: I think there was a point in these not requiring an Flix object.

In other words, they are independent of the specific flix compiler instance.

I think we should just pass the formatter implicitly.

**Comment by magnus-madsen** [comment]
If there are flix/maven dependencies which are downloaded, should we color them too?

**Comment by magnus-madsen** [comment]
Could you show a screenshot of a package that depends on the museum package? 

https://doc.flix.dev/packages.html#understanding-dependency-resolution

**Comment by sockmaster27** [comment]
billede

**Comment by magnus-madsen** [comment]
Lets keep the extra chars.

**Comment by magnus-madsen** [comment]
Conflicted, sorry

**Comment by sockmaster27** [comment]
RFM

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**